src/pushsource/_impl/backend/errata_source/errata_client.py

```diff
@@ -73,49 +107,197 @@ def get_raw_f(self, advisory_id):
                     "type": "errata-tool-call-start",
-                    "method": method_name,
+                    "method": method,
                     "advisory": advisory_id,
                 }
             },
         )
         try:
-            out = method(advisory_id)
+            out = self._do_call(method, advisory_id)
+
             LOG.debug(
                 "Errata Tool completed call %s(%s)",
-                method_name,
+                method,
                 advisory_id,
                 extra={
                     "event": {
                         "type": "errata-tool-call-end",
-                        "method": method_name,
+                        "method": method,
                         "advisory": advisory_id,
                     }
                 },
             )
+
             return out
         except:
             LOG.debug(
                 "Failed to call Errata Tool %s(%s)",
-                method_name,
+                method,
                 advisory_id,
                 extra={
                     "event": {
                         "type": "errata-tool-call-fail",
-                        "method": method_name,
+                        "method": method,
                         "advisory": advisory_id,
                     }
                 },
             )
             raise
+
+
+class ErrataHTTPClient(ErrataClientBase):
+    def __init__(self, threads, url, keytab_path: str = None,
+                 principal: str = None, **retry_args):
+        super().__init__(threads, url, **retry_args)
+
+        self.get_advisory_data = partial(
+            self._call_et, "/api/v1/push_metadata/ftp_paths/{id}.json"
+        )
+        self._get_advisory_cdn_metadata = partial(
+            self._call_et, "/api/v1/push_metadata/cdn_metadata/{id}.json"
+        )
+        self._get_advisory_cdn_file_list = partial(
+            self._call_et, "/api/v1/push_metadata/cdn_file_list/{id}.json"
+        )
+        self._get_advisory_cdn_docker_file_list = partial(
+            self._call_et,
+            "/api/v1/push_metadata/cdn_docker_file_list/{id}.json"
+        )
+        self._get_ftp_paths = partial(
+            self._call_et, "/api/v1/push_metadata/ftp_paths/{id}.json")
+
+        with tempfile.NamedTemporaryFile(prefix="ccache_pushsource_errata_",
+                                         delete=False) as file:
+            self.ccache_filename = file.name
+
+        if keytab_path:
+            self.keytab_path = keytab_path
+        else:
+            self.keytab_path = os.environ.get("PUSHSOURCE_ERRATA_KEYTAB_PATH",
+                                              None)
+
+        if principal:
+            self.principal = principal
+        else:
+            self.principal = os.environ.get("PUSHSOURCE_ERRATA_PRINCIPAL",
+                                            None)
+
+    def authenticate(self):
+        # Shunting like this for compatibility, since this class has probably
```

Comment:
While I don't have any problem with the design here, I'm not quite following what this comment means.
In case there is some misunderstanding, I want to point out that nothing outside of pushsource is expected to have access to the ErrataHTTPClient class.  It's not a part of the library's API.

Reply:
Is this the same for ErrataClient? I just wanted to make these classes interchangeable. I assumed ErrataHTTPClient could be accessed outside of PushSource and that it may have been used already. If that's not the case I can just put the kerberos ticket code under authentication

Reply:
Yes, it's the same.

If you look at https://github.com/release-engineering/pushsource/blob/master/src/pushsource/__init__.py , only those classes are the ones which can be accessed via `import pushsource`.

For everything else, there is no way to access it without writing `from pushsource._impl.(something)`, and if developers write that code they're supposed to understand from the naming that they're accessing something private & shouldn't be doing that